sentence_transformers/SentenceTransformer.py

```diff
@@ -723,14 +723,15 @@ def similarity_fn_name(
     def similarity(self, embeddings1: Tensor, embeddings2: Tensor) -> Tensor: ...
 
     @overload
-    def similarity(self, embeddings1: ndarray, embeddings2: ndarray) -> Tensor: ...
+    def similarity(self, embeddings1: npt.NDArray[np.float32], embeddings2: npt.NDArray[np.float32]) -> Tensor: ...
```

Comment:
Nice, looks to be the correct usage. I wasn't aware of these.
https://numpy.org/devdocs/reference/typing.html#numpy.typing.NDArray